microcosm_pubsub/errors.py

```diff
@@ -20,8 +20,10 @@ class Nack(Exception):
     and the SQS dead letter queue (DLQ) configuration, if any.
 
     """
-    def __init__(self, visibility_timeout_seconds):
+    def __init__(self, visibility_timeout_seconds, reason="", extra=None):
```

Comment:
Can reason be None instead of empty string? Then function signature would match `TTLExpired()`.

Reply:
Interesting that `reason` is required for `SkipMessage`. Nack should probably be like that someday.